rayon-core/src/configuration.rs

```diff
@@ -77,7 +58,7 @@ impl Configuration {
 
     /// Get the number of threads that will be used for the thread
     /// pool. See `set_num_threads` for more information.
-    pub fn num_threads(&self) -> Option<usize> {
+    pub fn num_threads(&self) -> usize {
         self.num_threads
```

Comment:
That's possible, but I'm not sure there's much benefit to doing it eagerly.  This isn't something I expect to get called repeatedly, and lazy computation seems simpler.